opal/corelib/string.rb

```diff
@@ -426,6 +426,21 @@ def end_with?(*suffixes)
     false
   end
 
+  def begin_with?(*suffixes)
+    %x{
+      for (var i = 0, length = suffixes.length; i < length; i++) {
+        var suffix = #{Opal.coerce_to(`suffixes[i]`, String, :to_str).to_s};
+
+        if (self.length >= suffix.length &&
+            self.substr(0, suffix.length) == suffix) {
+          return true;
+        }
+      }
+    }
+
+    false
+  end
+
```

Comment:
I think this can be omitted and `File.join` can use `#start_with?` instead.

http://www.rubydoc.info/stdlib/core/String#start_with%3F-instance_method